geode-core/src/test/java/org/apache/geode/internal/net/NioSslEngineTest.java

```diff
@@ -595,10 +602,17 @@ public int getNumberOfUnwraps() {
 
     @Override
     public SSLEngineResult wrap(ByteBuffer[] sources, int i, int i1, ByteBuffer destination) {
-      for (ByteBuffer source : sources) {
+      assert sources.length == 1;
+      final ByteBuffer source = sources[0];
+      final SSLEngineResult nextResult = nextResult();
+      try {
         destination.put(source);
+      } catch (final BufferOverflowException e) {
+        assertThat(BUFFER_OVERFLOW)
```

Comment:
`assertThat` should take the actual and `isEqualTo` should take the expected. If it is a constant you should use `isSameAs`.

Reply:
I initially did it the way you suggest. But it turns out the way this test works it makes more sense the way it is now.

See the test setup calls `TestSSLEngine.addReturnResult(SSLEngineResult... sslEngineResult)` to set the sequence of results to return from calls to the test engine's `wrap()` method. So the expectation in this case is that the state we would return, in this case `BUFFER_OVERFLOW`, matches the expectation set in `addReturnResult()`.

Hope you agree.

Reply:
I don't. I would treat the thing that can mutate, regardless of point of view, to be the actual value and the thing that can't, the constant, to be the expected.

Reply:
This logic is simulating part of the `SSLEngine`:

![image](https://user-images.githubusercontent.com/4002/162085410-c8315d63-89dd-4c3a-b60a-b952309d476f.png)

What we've "got" here is a buffer overflow situation, which the engine would interpret as a `BUFFER_OVERFLOW` status. What we "expect" (as expressed in the test setup's earlier call to `addReturnResult(SSLEngineResult... sslEngineResult)` is found in `nextResult.getStatus()`.

Reply:
The constant `BUFFER_OVERFLOW` literally can't change though. 🤷